gtwrap/pybind_wrapper.py

```diff
@@ -153,6 +155,51 @@ def _wrap_print(self, ret: str, method: parser.Method, cpp_class: str,
             suffix=suffix)
         return ret
 
+    def _wrap_dunder(self,
+                     method,
+                     cpp_class,
+                     prefix,
+                     suffix,
+                     method_suffix=""):
+        """
+        Wrap a Python double-underscore (dunder) method.
+
+        E.g. __len__() gets wrapped as `.def("__len__", [](gtsam::KeySet* self) {return self->size();})`
+
+        Supported methods are:
+        - __contains__(T x)
+        - __len__()
+        - __iter__()
+        """
+        py_method = method.name + method_suffix
+        args_names = method.args.names()
+        py_args_names = self._py_args_names(method.args)
+        args_signature_with_names = self._method_args_signature(method.args)
+
+        if method.name == 'len':
+            function_call = "return self->size();"
+        elif method.name == 'contains':
+            function_call = f"return self->find({method.args.args_list[0].name}) != self->end();"
+        elif method.name == 'iter':
+            function_call = "return py::make_iterator(self->begin(), self->end());"
```

Comment:
Would it make more sense for the .i files to specify the C++ function name instead of the python function name?  Eg instead of __len__, the function size would automatically generate the __len__ dunder?
My reasoning being that neither a python nor c++ user would ever actually call "__len__" since it's just a special-case syntax in python.
Downside being that size is much more ambiguous than __len__ so there may be false positives.
Just want to open for discussion.

Reply:
I was worried about false positives myself. For example, in the case of `__iter__`, I was thinking of originally making it just `iter`, but then C++ has its own definition of iterators and someone may decide to have a C++ method `iter() const` that returns a C++ iterator which would clash with this (note the special `py::make_iterator` definition).

Moreover, I think you're forgetting that the Matlab wrapper needs to be supported as well, so having explicit dunder annotations makes it obvious that these methods are for __Python only__ and won't be available in Matlab or any other language.

PS any reason you're requesting changes if there is no specific change being asked for?

Reply:
Fair.  Just for documenting this thought process, I think it clashes a bit with the vibe that a .i file should closely match the c++ interface (as opposed to the python interface), but reducing false-positives seems to be a reasonable tradeoff to me.